src/linalg/rankUpdate.jl

```diff
@@ -86,3 +88,14 @@ function rankUpdate!(C::HermOrSym{T,UniformBlockDiagonal{T}}, A::BlockedSparse{T
 end
 
 rankUpdate!(C::HermOrSym{T,Matrix{T}}, A::BlockedSparse{T}, α=true) where {T} = rankUpdate!(C, A.cscmat, α)
+
+function rankUpdate!(C::Diagonal{T, S}, A::Diagonal{T,S}, α=true, β=true) where {T, S}
+    # C := β*C + α*A'A
+
+    length(C.diag) == length(A.diag) ||
+        throw(DimensionMismatch("length(C.diag) ≠ length(A.diag)"))
+
+    for ii in 1:length(C.diag)
+        C.diag[ii] = β * C.diag[ii] + α * A.diag[ii] * A.diag[ii]
+    end
+end
```

Comment:
I think this can be written more concisely as
```julia
function rankUpdate!(C::Diagonal{T,S}, A::Diagonal{T,S}, α::Number=true, β::Number=true) where {T, S}
    C.diag .= β .* C.diag .+ α .* abs2.(A.diag)
    C
end
```